public/app/features/templating/template_srv.ts

```diff
@@ -347,6 +348,51 @@ export class TemplateSrv implements BaseTemplateSrv {
     });
   }
 
+  getAllVariablesInTarget(target: string, scopedVars: ScopedVars, format?: string | Function): VariableMap {
+    const values: VariableMap = {};
+
+    this.replaceInVariableRegex(target, (match, var1, var2, fmt2, var3, fieldPath, fmt3) => {
+      const variableName = var1 || var2 || var3;
+      const variableDisplayName =
+        var1 || var2 || (var3 !== undefined && fieldPath !== undefined) ? `${var3}.${fieldPath}` : var3;
+      const fmt = fmt2 || fmt3 || format;
+      const value = this.getVariableValue(variableName, fieldPath, scopedVars);
+      if (value !== null && value !== undefined) {
+        const variable = this.getVariableAtIndex(variableName);
+        const text = this.getVariableText(variableName, value, scopedVars);
+        values[variableDisplayName] = this.formatValue(value, fmt, variable, text);
+      } else {
+        values[variableDisplayName] = undefined;
+      }
+
+      // Don't care about the result anyway
+      return '';
+    });
+
+    return values;
+  }
```

Comment:
@gelicia 

Not sure I don't really understand this implementation (and use case), and lacks tests 

I am sorry, I think I will have to revert this PR until this can be more explained (and has some tests)

I am currently rewriting the whole variable system & interpolation system in scenes. 

Also adding new functions to templateSrv (exposed to plugins) needs some review from the dashboard squad.